resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/registry/SegmentNode.java

```diff
@@ -101,7 +101,23 @@ public ResourceInvoker match(HttpRequest request, int start)
                }
                else
                {
-                  String substring = path.substring(0, length);
+                  // must find the end of the matched pattern
+                  // and get the substring from 1st char thru end
+                  // of matched chars
+                  Pattern p = expression.getPattern();
+                  Matcher m = p.matcher(path);
+                  m.region(start, path.length());
+                  String substring = path;
+                  while(m.find()) {

```

Comment:
@asoldano @ronsigal @rsearls I've done all my local testings (include resteasy-js testings) and didn't find any problem with this patch. So I suggest we merge it.